### PR TITLE
app-content-pages: Responsive styling and mobile dropdown

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-content-pages

## Linked Issue and/or Talk Post
Closes: https://github.com/zooniverse/front-end-monorepo/issues/3591
Closes: https://github.com/zooniverse/front-end-monorepo/issues/5476

Figma design: https://www.figma.com/file/zN9InDg8CTk2vnWp406moK/About-Pages
Project board: https://github.com/orgs/zooniverse/projects/40

## Describe your changes

- Built a DropdownNav component for smaller viewport widths (sticky).
- Built a responsive grid in Publications and Team. It aligns the main content `center`, fits the sidebar into the left-hand space until screens <72rem in width. On screens <72rem, the sidebar disappears and the dropdown nav is used instead.
- Fixed a bug where the psuedo elements `::before` and `::after` the page layout.js caused awkward horizontal scrolling on screens <90rem.
- Reduced the font size of labels in the sidebar to 14px, and matched the font-size of headers to the Figma design.
- I put several shared `styled-components` into a file called SharedStyledComponents. They are shared in the Teams and Publications pages, but will not necessarily be used for Resources, FAQ, or the root About page, so I didn't put those components directly in layout.js.
- I created a MaxWidthContent component that's a simple Box with a max-width of 45rem. This will be used through all /about and /get-involved pages.

Here are some updated screenshots after multiple design review (Nov 13):
team page
publications page

Mobile:
publications mobile

team mobile

dropdown mobile

## How to Review

There are a lot of lines of code changed in this PR, but it is almost exclusively design related. I took the route of more css properties in `styled` components rather than wrestle with Grommet theming primarily due to the responsive grid breakpoint being 72rem rather than the 'small' vs 'medium' default Grommet breakpoint.

- Run this branch locally and visit https://local.zooniverse.org:3000/about/publications and https://local.zooniverse.org:3000/about/team
- You can also check out the app-content-pages storybook

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by goplayoutside3** [comment]
Making this draft for a few days while I add the following revisions:
- [ ] Style the DropdownNav like the teal dropdown in the ProjectHeader (app-project)
- [ ] Give the mobile dropbutton a drop-shadow and 20px margin-top
- [ ] Considering putting some shared design elements into a /shared/DesignElements folder for use on multiple pages. There are a few common `styled` components in Teams.js and Publications.js

Next design review with Sean will be Monday Nov 6!

**Comment by goplayoutside3** [comment]
This branch wasn't able to build and deploy to `fe-project-branch`: https://github.com/zooniverse/front-end-monorepo/actions/runs/6775543123/job/18415054771. Although we can't preview app-content-pages via the `fe-project-branch` url, I think the same build fail was seen when https://github.com/zooniverse/front-end-monorepo/pull/5576 was first merged into this branch: https://github.com/zooniverse/front-end-monorepo/actions/runs/6775448960/job/18414750944?pr=5578.

I'm going to try a few suggestions from https://nextjs.org/docs/messages/prerender-error, and then do the branch deploy again this afternoon.

**Comment by goplayoutside3** [comment]
This PR has been through two design reviews with Sean, and now needs a dev-review, sanity check that the pages look as expected when run in a browser or storybook.

File: packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -17,16 +17,18 @@ const StyledButton = styled(Button)`
...
-      background: #addde0; // accent-1
-      font-weight: bold;
+      background: ${props.theme.global.colors['accent-1']};
     `}
 
-  &:hover {
-    font-weight: bold;
+  &:hover, :focus {
+    > span {
+      font-weight: bold;
```
Comment (mcbouslog) [inline packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js:30]: This is causing the component width to shift/expand when the largest item is hovered (i.e. SOCIAL SCIENCES or ADLER PLANETARIUM) in a noticeable way. Not blocking, and I'm not sure what the solution is 🤔 . Most noticeable in the middle width (~ 1200px).

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js [hunk 2]
```diff
@@ -1,30 +1,8 @@
...
 import { Anchor, Box, Text } from 'grommet'
 import { string } from 'prop-types'
-import styled from 'styled-components'
-
-const StyledBox = styled(Box)`
-  border-radius: 5px;
-  overflow: hidden;
-`
-
-const Placeholder = ({ displayString }) => {
```
Comment (mcbouslog) [inline packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:11]: Optionally, with this moving to Project.js maybe the related publications-publication--placeholders story can be removed and a placeholder related story could be added to the Project.stories.

**Review by mcbouslog — COMMENTED** [review]
This is looking great! Comments inline are not blocking.

Not approving just yet because have one question - on mobile when selecting an option from dropdown, it covers the related selection title, and the dropdown doesn't show what was selected, so as a user I'm not sure where I am. 

Here's an attempted capture with BrowserStack -
https://github.com/zooniverse/front-end-monorepo/assets/12118751/cae17b30-484a-4e55-b3ab-8af1e0852c9d

I think that might be worth revisiting, but open to explanation otherwise if that's intended.

File: packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js:30]: I'm going to leave as is for now. I tried out a couple of quick ideas, but can't find a solution at the moment.

File: packages/app-content-pages/src/shared/components/DropdownNav/DropdownNav.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        position: absolute;
+        bottom: 0;
+        left: 0;
+        background: ${props.theme.global.colors.brand};
+        height: 50%;
+        width: 100%;
+        z-index: -1;
+      }
+    `}
+`
```
[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/app-content-pages/src/shared/components/DropdownNav/DropdownNav.js:59]: Yes your'e right! I'll fix. I actually don't want the background to change on focus either, so I'll remove that rule in favor of just hover or open.

File: packages/app-content-pages/src/shared/components/DropdownNav/DropdownNav.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  &:hover {
+    ${props => css`
+      background: ${props.theme.global.colors['accent-1']};
+    `}
+
+  ${props =>
+    props.open &&
+    css`
+      box-shadow: none;
+      background: #addde0; // accent-1
```
[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/app-content-pages/src/shared/components/DropdownNav/DropdownNav.js:46]: Yes I'll do that 👍

File: packages/app-content-pages/src/shared/components/DropdownNav/DropdownNav.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    >
+      <Box
+        align='center'
+        direction='row'
+        gap='xsmall'
+        justify='center'
+        pad={{ horizontal: '20px', vertical: '10px' }}
+        round='small'
+      >
+        <SpacedText weight='bold' color={{ light: 'brand', dark: '' }}>
```
[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/app-content-pages/src/shared/components/DropdownNav/DropdownNav.js:140]: This style is a copy over from the project About Pages' styling. It'll be easier to test after https://github.com/zooniverse/front-end-monorepo/issues/5571 is addressed.

**Comment by goplayoutside3** [comment]
> on mobile when selecting an option from dropdown, it covers the related selection title, and the dropdown doesn't show what was selected, so as a user I'm not sure where I am.

This happens because the link brings the corresponding h2 to the top of the viewport, but the dropdown nav is sticky and the space it takes up covers the h2. I've got a fix for this, but also noticed that the dropdown nav's highlight is not working properly, so will fix that too and then re-request review.

**Review by mcbouslog — APPROVED** [review]
Changes look good ✅ . Confirmed in mobile that nav selection doesn't cover section title 👍 .  Responses to in-line comments sound good. 🚀

**Pull request merged**